sdk/typespec/typespec_derive/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+extern crate proc_macro;
+
+use proc_macro2::{Span, TokenStream};
+use syn::parse::ParseStream;
+use syn::spanned::Spanned;
+use syn::{Attribute, DeriveInput, LitStr, Meta, Path};
+
+/// Derive macro for implementing `Model` trait.
+/// Deriving this trait allows a type to be deserialized from an HTTP response body.
```

Comment:
Nit: I'd put a blank line in here or it all gets mushed together and harder to read.

There's a lint coming soon to nightly that will complain about long "first lines" as well.